jwst/resample/resample_step.py

```diff
@@ -225,20 +226,16 @@ def _set_spec_defaults(self):
         # Force definition of good bits
         kwargs['good_bits'] = GOOD_BITS
 
-        if kwargs['pixfrac'] is None:
-            kwargs['pixfrac'] = 1.0
-        if kwargs['kernel'] is None:
-            kwargs['kernel'] = 'square'
-        if kwargs['fillval'] is None:
-            kwargs['fillval'] = 'INDEF'
-        if kwargs['weight_type'] is None:
-            kwargs['weight_type'] = 'ivm'
+        kwargs['pixfrac'] = self.pixfrac
+        kwargs['kernel'] = str(self.kernel)
+        kwargs['fillval'] = str(self.fillval)
```

Comment:
Should there be similar lines in resample_spec for pixfrac, kernel, and fillval as what you've added for weight_type? I don't know if they get called/used in the spectral version, but I do at least see the pixfrac value is added to the output product in resample_spec_step's _process_multislit().

Reply:
The code you're seeing here is the `_set_spec_defaults` function, which lives in the resample_step module, but is called by both resample and resample_spec. So in fact the identical settings are being used for both.

Reply:
Sorry, I didn't read closely enough, though a question still remains - upon looking more closely, it looks like get_drizpars() tries to deal with the combination of user vs. reference values for these keywords. But in the case where drizpars exists, would pixfrac still get overridden to a value of 1.0 because it's the default spec value, even though the user has not specified it? The extract_1d bkg_fit issue has me tuned into this problem now.

Reply:
From a quick test with NIRCam's drizpars file, it looks like the reference values are being ignored because the default spec values are not None.